File: leetcode/go/1310.xor-queries-of-a-subarray.go
func xorQueries(arr []int, queries [][]int) []int {
    xors := make([]int, len(arr) + 1)
    for i := range len(arr) {
        xors[i+1] = xors[i] ^ arr[i]
    }
    ans := make([]int, len(queries))
    for i, q := range queries {
        ans[i] = xors[q[1] + 1] ^ xors[q[0]]
    }
    return ans
}

File: leetcode/rust-leetcode/src/1310.xor-queries-of-a-subarray.rs
impl Solution {
    pub fn xor_queries(arr: Vec<i32>, queries: Vec<Vec<i32>>) -> Vec<i32> {
        let mut xors = vec![0; arr.len()+1];
        for i in 0..arr.len() {
            xors[i+1] = xors[i] ^ arr[i];
        }
        let mut ans = vec![0; queries.len()];
        for i in 0..queries.len() {
            ans[i] = xors[queries[i][1] as usize + 1] ^ xors[queries[i][0] as usize];
        }
        ans
    }
}

